test/optimizer/stats_calculator_test.cpp

```diff
@@ -0,0 +1,557 @@
+    table_1_col_oid_ = accessor_->GetSchema(table_oid_1_).GetColumns().at(0).Oid();
+
+    NOISEPAGE_ASSERT(accessor_->GetSchema(table_oid_2_).GetColumns().size() == 2, "Table should have 2 column");
+    table_2_col_1_oid_ = accessor_->GetSchema(table_oid_2_).GetColumns().at(0).Oid();
+    table_2_col_2_oid_ = accessor_->GetSchema(table_oid_2_).GetColumns().at(1).Oid();
+  }
+};
+
+// NOLINTNEXTLINE
+TEST_F(StatsCalculatorTests, TestLogicalGet) {
+  RunQuery("INSERT INTO " + table_name_1_ + " VALUES(1), (NULL), (3);");
+  RunQuery("ANALYZE " + table_name_1_ + ";");
+  txn_manager_->Commit(test_txn_, transaction::TransactionUtil::EmptyCallback, nullptr);
+  test_txn_ = txn_manager_->BeginTransaction();
+
+  Operator logical_get =
+      LogicalGet::Make(test_db_oid_, table_oid_1_, {}, table_name_1_, false).RegisterWithTxnContext(test_txn_);
+  GroupExpression *gexpr = new GroupExpression(logical_get, {}, test_txn_);
+  gexpr->SetGroupID(group_id_t(1));
+  context_.GetMemo().InsertExpression(gexpr, false);
+
+  parser::ColumnValueExpression col_a(table_name_1_, table_1_col_1_name_, test_db_oid_, table_oid_1_, table_1_col_oid_,
+                                      type::TypeId::INTEGER);
+  ExprSet required_cols;
+  required_cols.emplace(&col_a);
+
+  stats_calculator_.CalculateStats(gexpr, required_cols, &context_);
+
+  auto *root_group = context_.GetMemo().GetGroupByID(gexpr->GetGroupID());
+  EXPECT_EQ(root_group->GetNumRows(), 3);
+  EXPECT_TRUE(root_group->HasColumnStats(table_name_1_ + "." + table_1_col_1_name_));
+}
+
+// NOLINTNEXTLINE
+TEST_F(StatsCalculatorTests, TestInvalidLogicalGet) {
+  Operator logical_get =
+      LogicalGet::Make(test_db_oid_, catalog::INVALID_TABLE_OID, {}, "", false).RegisterWithTxnContext(test_txn_);
+  GroupExpression *gexpr = new GroupExpression(logical_get, {}, test_txn_);
+  gexpr->SetGroupID(group_id_t(1));
+  context_.GetMemo().InsertExpression(gexpr, false);
+
+  parser::ColumnValueExpression col_a(table_name_1_, table_1_col_1_name_, test_db_oid_, table_oid_1_, table_1_col_oid_,
+                                      type::TypeId::INTEGER);
+  ExprSet required_cols;
+  required_cols.emplace(&col_a);
+
+  stats_calculator_.CalculateStats(gexpr, required_cols, &context_);
+
+  auto *root_group = context_.GetMemo().GetGroupByID(gexpr->GetGroupID());
+  EXPECT_EQ(root_group->GetNumRows(), -1);
+  EXPECT_FALSE(root_group->HasColumnStats(table_name_1_ + "." + table_1_col_1_name_));
+}
+
+// NOLINTNEXTLINE
+TEST_F(StatsCalculatorTests, TestNotPredicate) {
+  RunQuery("INSERT INTO " + table_name_1_ + " VALUES(1), (NULL), (3);");
+  RunQuery("ANALYZE " + table_name_1_ + ";");
+  txn_manager_->Commit(test_txn_, transaction::TransactionUtil::EmptyCallback, nullptr);
+  test_txn_ = txn_manager_->BeginTransaction();
+
+  parser::ColumnValueExpression col_a(table_name_1_, table_1_col_1_name_, test_db_oid_, table_oid_1_, table_1_col_oid_,
```

Comment:
Can you add a comment on what is the predicate that you're constructing? The same applies to all the other tests with predicates and joins. Otherwise, people really have to look through all the construction code to figure out what's tested.

Reply:
Done